app/models/manageiq/providers/redhat/infra_manager/api_integration.rb

```diff
@@ -19,32 +20,38 @@ def connect(options = {})
     unless options[:skip_supported_api_validation] || supports_the_api_version?(version)
       raise "version #{version} of the api is not supported by the provider"
     end
-    # If there is API path stored in the endpoints table and use it:
+
+    # If the API path is stored in the endpoints table then use it:
     path = options[:path] || default_endpoint.path
     _log.info("Using stored API path '#{path}'.") unless path.blank?
 
-    server   = options[:ip] || address
-    port     = options[:port] || self.port
-    username = options[:user] || authentication_userid(options[:auth_type])
-    password = options[:pass] || authentication_password(options[:auth_type])
-    service  = options[:service] || "Service"
+    # Prepare the options to call the method that creates the actual connection:
+    connect_options = {
+      :server     => options[:ip] || address,
+      :port       => options[:port] || self.port,
+      :username   => options[:user] || authentication_userid(options[:auth_type]),
+      :password   => options[:pass] || authentication_password(options[:auth_type]),
+      :service    => options[:service] || "Service",
+      :verify_ssl => default_endpoint.verify_ssl,
+      :ca_certs   => default_endpoint.certificate_authority
+    }
 
     # Starting with version 4 of oVirt authentication doesn't work when using directly the IP address, it requires
     # the fully qualified host name, so if we received an IP address we try to convert it into the corresponding
     # host name:
     if resolve_ip_addresses?
-      resolved = resolve_ip_address(server)
-      if resolved != server
-        _log.info("IP address '#{server}' has been resolved to host name '#{resolved}'.")
+      resolved = resolve_ip_address(connect_options[:server])
+      if resolved != connect_options[:server]
+        _log.info("IP address '#{connect_options[:server]}' has been resolved to host name '#{resolved}'.")
         default_endpoint.hostname = resolved
-        server = resolved
+        connect_options[:server] = resolved
```

Comment:
You are right, reverse DNS lookups are secure/insecure as the DNS system itself. If the DNS system has been compromised, this will allow that kind of attacks, assuming that the user knows and trusts the CA certificate presented by the `evil.mordor` site. I hope that users understand that one of the key points of points of the security of TLS is checking CA certificates. If users just blindly take CA certificates from untrusted sources, and they paste them into the text box without checking them, then all the security goes away, with or without reverse DNS lookups.

There may be legitimate cases where the user doesn't want to use reverse DNS lookups. Those will be rare, as oVirt doesn't accept IP addresses, only fully qualified names. For those rage cases where the user really wants to avoid reverse DNS lookups, the oVirt provider has a `resolve_ip_addresses` parameter (default to `true`) and the oVirt server also has configuration parameters to allow IP addresses in addition to fully qualified names.

The object is saved to the database when doing inventory refresh.